util/java/libtiled-java/src/tiled/core/MapObject.java

```diff
@@ -151,24 +152,62 @@ public Image getImage(double zoom) {
         return scaledImage;
     }
 
-    public int getX() {
+    public double getX() {
         return bounds.x;
     }
 
-    public void setX(int x) {
+    public void setX(double x) {
         bounds.x = x;
     }
 
-    public int getY() {
+    public double getY() {
         return bounds.y;
     }
 
-    public void setY(int y) {
+    public void setY(double y) {
         bounds.y = y;
     }
 
-    public void translate(int dx, int dy) {
-        bounds.translate(dx, dy);
+    public void translate(double dx, double dy) {
+        double oldv = bounds.x;
+        double newv = oldv + dx;
+        if (dx < 0) {
+            if (newv > oldv) {
+                if (bounds.width >= 0) {
+                    bounds.width += newv - Double.MIN_VALUE;
+                }
+                newv = Double.MIN_VALUE;
+            }
+        } else {
+            if (newv < oldv) {
+                if (bounds.width >= 0) {
+                    bounds.width += newv - Double.MAX_VALUE;
+                    if (bounds.width < 0) bounds.width = Double.MAX_VALUE;
+                }
+                newv = Double.MAX_VALUE;
+            }
+        }
+        bounds.x = newv;
+
+        oldv = bounds.y;
+        newv = oldv + dy;
+        if (dy < 0) {
+            if (newv > oldv) {
+                if (bounds.height >= 0) {
+                    bounds.height += newv - Double.MIN_VALUE;
+                }
+                newv = Double.MIN_VALUE;
+            }
+        } else {
+            if (newv < oldv) {
+                if (bounds.height >= 0) {
+                    bounds.height += newv - Double.MAX_VALUE;
+                    if (bounds.height < 0) bounds.height = Double.MAX_VALUE;
+                }
+                newv = Double.MAX_VALUE;
+            }
+        }
+        bounds.y = newv;
```

Comment:
Note that doubles don't wrap around like integers do, so I don't think anything more meaningful can really be done here.